src/main/java/com/arangodb/internal/velocystream/VstCommunication.java

```diff
@@ -92,6 +92,13 @@ protected synchronized C connect(final HostHandle hostHandle, final AccessType a
                         tryAuthenticate(connection);
                     }
                     hostHandler.confirm();
+                    if (!connection.isOpen()) {
+                        // see https://github.com/arangodb/arangodb-java-driver/issues/384
+                        connection.close();
```

Comment:
```suggestion
```
The connection here should be already closed.

Reply:
Thanks. Fixed!
(just being over-protective not to leave anything behind) :)